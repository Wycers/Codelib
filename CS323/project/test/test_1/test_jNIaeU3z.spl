int a,b,c;
float m,n;
int fibo(int a)
{
    if (a == 1 || a == 2) return 1;
    return fibo(a-1)+fibo(a-2);
}

int main()
{
    int m,n,i;
    m = read();
    i=1;
    while(i<=m)
    {
        if(m > n){
          n = fibo(i);
          write(n);
          i=i+1;
        }
    }
    return 1;
}