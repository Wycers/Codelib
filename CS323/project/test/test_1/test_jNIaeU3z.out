Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      ExtDecList (1)
        VarDec (1)
          ID: a
        COMMA
        ExtDecList (1)
          VarDec (1)
            ID: b
          COMMA
          ExtDecList (1)
            VarDec (1)
              ID: c
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          TYPE: float
        ExtDecList (2)
          VarDec (2)
            ID: m
          COMMA
          ExtDecList (2)
            VarDec (2)
              ID: n
        SEMI
      ExtDefList (3)
        ExtDef (3)
          Specifier (3)
            TYPE: int
          FunDec (3)
            ID: fibo
            LP
            VarList (3)
              ParamDec (3)
                Specifier (3)
                  TYPE: int
                VarDec (3)
                  ID: a
            RP
          CompSt (4)
            LC
            StmtList (5)
              Stmt (5)
                IF
                LP
                Exp (5)
                  Exp (5)
                    Exp (5)
                      ID: a
                    EQ
                    Exp (5)
                      INT: 1
                  OR
                  Exp (5)
                    Exp (5)
                      ID: a
                    EQ
                    Exp (5)
                      INT: 2
                RP
                Stmt (5)
                  RETURN
                  Exp (5)
                    INT: 1
                  SEMI
              StmtList (6)
                Stmt (6)
                  RETURN
                  Exp (6)
                    Exp (6)
                      ID: fibo
                      LP
                      Args (6)
                        Exp (6)
                          Exp (6)
                            ID: a
                          MINUS
                          Exp (6)
                            INT: 1
                      RP
                    PLUS
                    Exp (6)
                      ID: fibo
                      LP
                      Args (6)
                        Exp (6)
                          Exp (6)
                            ID: a
                          MINUS
                          Exp (6)
                            INT: 2
                      RP
                  SEMI
            RC
        ExtDefList (9)
          ExtDef (9)
            Specifier (9)
              TYPE: int
            FunDec (9)
              ID: main
              LP
              RP
            CompSt (10)
              LC
              DefList (11)
                Def (11)
                  Specifier (11)
                    TYPE: int
                  DecList (11)
                    Dec (11)
                      VarDec (11)
                        ID: m
                    COMMA
                    DecList (11)
                      Dec (11)
                        VarDec (11)
                          ID: n
                      COMMA
                      DecList (11)
                        Dec (11)
                          VarDec (11)
                            ID: i
                  SEMI
              StmtList (12)
                Stmt (12)
                  Exp (12)
                    Exp (12)
                      ID: m
                    ASSIGN
                    Exp (12)
                      ID: read
                      LP
                      RP
                  SEMI
                StmtList (13)
                  Stmt (13)
                    Exp (13)
                      Exp (13)
                        ID: i
                      ASSIGN
                      Exp (13)
                        INT: 1
                    SEMI
                  StmtList (14)
                    Stmt (14)
                      WHILE
                      LP
                      Exp (14)
                        Exp (14)
                          ID: i
                        LE
                        Exp (14)
                          ID: m
                      RP
                      Stmt (15)
                        CompSt (15)
                          LC
                          StmtList (16)
                            Stmt (16)
                              IF
                              LP
                              Exp (16)
                                Exp (16)
                                  ID: m
                                GT
                                Exp (16)
                                  ID: n
                              RP
                              Stmt (16)
                                CompSt (16)
                                  LC
                                  StmtList (17)
                                    Stmt (17)
                                      Exp (17)
                                        Exp (17)
                                          ID: n
                                        ASSIGN
                                        Exp (17)
                                          ID: fibo
                                          LP
                                          Args (17)
                                            Exp (17)
                                              ID: i
                                          RP
                                      SEMI
                                    StmtList (18)
                                      Stmt (18)
                                        Exp (18)
                                          ID: write
                                          LP
                                          Args (18)
                                            Exp (18)
                                              ID: n
                                          RP
                                        SEMI
                                      StmtList (19)
                                        Stmt (19)
                                          Exp (19)
                                            Exp (19)
                                              ID: i
                                            ASSIGN
                                            Exp (19)
                                              Exp (19)
                                                ID: i
                                              PLUS
                                              Exp (19)
                                                INT: 1
                                          SEMI
                                  RC
                          RC
                    StmtList (22)
                      Stmt (22)
                        RETURN
                        Exp (22)
                          INT: 1
                        SEMI
              RC
